pandas/core/indexes/base.py

```diff
@@ -814,8 +813,10 @@ def take(self, indices, axis=0, allow_fill=True, fill_value=None, **kwargs):
             )
         else:
             if allow_fill and fill_value is not None:
-                msg = "Unable to fill values because {0} cannot contain NA"
-                raise ValueError(msg.format(self.__class__.__name__))
+                cls_name = self.__class__.__name__
```

Comment:
we're now using type() instead of `__class__`, see https://github.com/pandas-dev/pandas/pull/29775#discussion_r349407678